src/gather.jl

```diff
@@ -0,0 +1,25 @@
+function gather_kernel!(dst, src, idx, max_idx, max_dims_idx, dims_size)
+    index = threadIdx().x + (blockIdx().x - 1) * blockDim().x
+
+    @inbounds if index <= max_idx
+        j, k = divrem(index-1, max_dims_idx)
+        dims_i = CartesianIndices(dims_size)[k+1]
+        dst[index] = src[Tuple(dims_i)..., idx[j+1]...]
+    end
+    return nothing
+end
+
+function NNlib.gather!(dst::AnyCuArray, src::AnyCuArray, idx::AnyCuArray)
+    dims = NNlib._check_dims(src, dst, idx)
```

Comment:
can you copy the definition over here `_check_dims`? Better not rely on internals as this is a separate library (also the name `_check_dims` is too generic, we should change it to `gather_check_dims`

Reply:
Sure! However, the `gather_check_dims` is generic. It is available for both cpu and gpu. Should I define it in NNlibCUDA? Or it is just fine to put it here?